operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Reconciler.java

```diff
@@ -14,6 +20,78 @@ public interface Reconciler<R extends HasMetadata> {
    * @return UpdateControl to manage updates on the custom resource (usually the status) after
    *         reconciliation.
    */
-  UpdateControl<R> reconcile(R resource, Context<R> context) throws Exception;
+  UpdateControl<P> reconcile(P resource, Context<P> context) throws Exception;
+
 
+  /**
+   * Prepares a map of {@link EventSource} implementations keyed by the name with which they need to
+   * be registered by the SDK.
+   *
+   * @param context a {@link EventSourceContext} providing access to information useful to event
+   *        sources
+   * @return a map of event sources to register
+   */
+  default Map<String, EventSource> prepareEventSources(EventSourceContext<P> context) {
+    return Map.of();
+  }
+
+  /**
+   * Utility method to easily create map with generated name for event sources. This is for the use
+   * case when the event sources are not access explicitly by name in the reconciler.
+   *
+   * @param eventSources to name
+   * @return even source with default names
+   */
+  static Map<String, EventSource> nameEventSources(EventSource... eventSources) {
```

Comment:
I would prefer that we move all these methods to a different utilty class. They have nothing to do with Reconciler itself and pollute its API, imo.

Reply:
Will move it for now, was thinking to also have have EventSources "Named" like now with the Dependent Resources. Will make a separate PR. That would make these methods basically obsolete.

Reply:
Added `EventSourceUtils`.